Conduit listing

sirajulmcb opened the pull request:
List all the conduits of the user in the dashboard once authenticated.

vakareeralil commented:
Found a couple of issues. 
1. After signup, the redirection to login page fails due to error with mapping. I wonder how the home page gets executed here when I am taken back to the login page. The conduit api call also fails as well. 
2. The case when no conduit is available is not handled. In this case, we should be having a button which gives them a proper message and a button to start creating one.

sirajulmcb commented:
I've fixed the empty conduits issue. Create is yet to be added.
I've added Edit function for existing conduit, which can be extended to create in the upcoming PR

Not able to figure out why Home is rendering when navigating to the /login route. It doesn't occur when navigating to /signup though.

I'll have to debug more on it.

vakareeralil commented on praas-app-ionic/src/pages/Home/index.tsx:
```diff
@@ -1,21 +1,85 @@
-const mapStateToProps = ({ user }: any) => ({
-  user,
+const mapStateToProps = ({ conduit }: any) => ({
+  conduits: conduit?.list?.conduits || [],
```
Also I see conduit.list is always an array at the worst case. Why do we need optional chaining here?

sirajulmcb replied:
I had some error previously, when there was an issue with the routing. Now it seems to be working without optional chaining in either cases when the list is empty or filled.

sirajulmcb commented on praas-app-ionic/src/pages/Home/style.scss:
```diff
@@ -0,0 +1,32 @@
+.home-page {
+  max-width: 1200px;
```
I'd prefer to use a width upto `1200px`, so it helps you to target bigger properly.
https://1200px.com/

Yes, we do use flexbox, as much as possible everywhere, but not a fan of using it for everything 😉 old CSS props can sometimes do good things 😄 

We'd need to create a new stylesheet and define reusable classes in there, but could be a different exercise, once we reach feature parity.

vakareeralil replied:
Lets make it a part of a new ticket then

sirajulmcb commented on praas-app-ionic/src/pages/Home/index.tsx:
```diff
@@ -1,21 +1,85 @@
 interface Props {
-  user: any;
+  conduits: Array<any>;
+  listConduits: () => void;
```
`Function`, `String`, `Number`, `Boolean` etc. doesn't seem to be valid primitives typescript.
`()=> void` would be more explicit about the input and output types of a function. `Function` seems to be more generic, similar to using `any`.
https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html

sirajulmcb commented on praas-app-ionic/src/pages/Home/style.scss:
```diff
@@ -0,0 +1,32 @@
+
+  .icon {
+    &.delete {
+      color: #ff0000;
```
Done
